cpp/oneapi/dal/algo/triangle_counting/detail/relabel_kernel.hpp

```diff
@@ -0,0 +1,120 @@
+/*******************************************************************************
+* Copyright 2020-2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/graph/detail/undirected_adjacency_vector_graph_impl.hpp"
+
+namespace oneapi::dal::preview::triangle_counting::detail {
+
+ONEDAL_EXPORT void sort_ids_by_degree(const dal::detail::host_policy& policy,
+                                      const std::int32_t* degrees,
+                                      std::pair<std::int32_t, std::size_t>* degree_id_pairs,
+                                      std::int64_t vertex_count);
```

Comment:
why size_t

Reply:
Size_t because size_t was used in TC protype and copied here with the same type. It will be good to make this  int32_t but better to make this in another PR +  required tests to be covered.